main.py

```diff
@@ -247,6 +247,7 @@ def draw_next_shape(shape, surface):
         for j, column in enumerate(row):

```

Comment:
What does enumerate do here?

Reply:
So in this line it looks like the enumerate function searches a row in the shape array.  This might look like ['..00..'].
As it looks at each element, if it sees a 0, it will fill that space in with a rectangle piece.  If there is a ., it will continue to the next iteration.  In this way one row of the shape can be drawn.  Doing this over all of the rows will cause a shape to take form.  The enumerate function is a convenient way to track the index of an item in an array while cycling through it.    Here is a video example explaining the function.  https://www.youtube.com/watch?v=-MZiQaNI0QA